spec/features/editing_images/upload_image_modal_requirements_spec.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+RSpec.feature "Upload image in a modal with requirements issues", js: true do
+  scenario do
+    given_there_is_an_edition
+    when_i_go_to_edit_the_edition
+    and_i_click_to_insert_an_image
+    and_i_upload_an_invalid_image
+    then_i_should_see_an_error
+  end
+
+  def given_there_is_an_edition
+    body_field = build(:field, id: "body", type: "govspeak")
+    document_type = build(:document_type, contents: [body_field], images: true)
+    @edition = create(:edition, document_type_id: document_type.id)
+  end
+
+  def when_i_go_to_edit_the_edition
+    visit edit_document_path(@edition.document)
+  end
+
+  def and_i_click_to_insert_an_image
+    within(".app-c-markdown-editor") do
+      find("markdown-toolbar details").click
+      click_on "Image"
+    end
+  end
+
+  def and_i_upload_an_invalid_image
+    find('form input[type="file"]').set(Rails.root.join(file_fixture("text-file.txt")))
```

Comment:
I wrote a comment on this before but it doesn't seem to be here so apologies if it's come through twice - I'm assuming I just lost it in the review

I notice we didn't need the Rails.root.join on https://github.com/alphagov/content-publisher/blob/7386e790f9669c4bb3fbd9e0ea7d15fd40b6494c/spec/features/editing_images/upload_image_requirements_spec.rb#L22 so wondering if this is needed or special.

Reply:
Yes, it was strange - I encountered this and forgot to come back to it. I've added another commit to explain, but it seems like absolute paths are required for JS feature specs.

Reply:
Oh that's a pain - thanks for the commit explanation